src/System.CommandLine/Help/HelpBuilder.cs

```diff
@@ -204,103 +206,199 @@ private void AppendDescription(string description)
+        {
+            foreach (var row in table)
+                AppendRow(row, columnWidths);
+        }
+
+        /// <summary>
+        /// Writes a row of strings to the console with columns of the given width.
+        /// </summary>
+        /// <param name="row">The row of elements to write.</param>
+        /// <param name="columnWidths">The width of each column of the table.</param>
+        protected virtual void AppendRow(IEnumerable<string> row, IReadOnlyList<int> columnWidths)
+        {
+            var split = row.Select((element, index) => SplitText(element, columnWidths[index])).ToArray();
+            var longest = split.Max(lines => lines.Count);
+            for (int line = 0; line < longest; ++line)
             {
-                AppendLine(descriptionLine, offset);
+                var columnStart = 0;
+                var appended = 0;
+                AppendPadding(CurrentIndentation);
+
+                for (int column = 0; column < split.Length; ++column)
+                {
+                    var lines = split[column];
+                    if (line < lines.Count)
+                    {
+                        var text = lines[line];
+                        if (!string.IsNullOrEmpty(text))
+                        {
+                            var offset = columnStart - appended;
+                            AppendText(text, offset);
+                            appended += offset + text.Length;
+                        }
+                    }
+                    columnStart += columnWidths[column] + ColumnGutter;
+                }
+                AppendBlankLine();
             }
         }
 
         /// <summary>
-        /// Takes a string of text and breaks it into lines of <see cref="maxLength"/>
-        /// characters. This does not preserve any formatting of the incoming text.
+        /// Takes a string of text and breaks it into lines of <paramref name="width"/>
+        /// characters. Whitespace formatting of the incoming text is removed.
         /// </summary>
-        /// <param name="text">Text content to split into writable lines</param>
-        /// <param name="maxLength">
-        /// Maximum number of characters allowed for writing the supplied <see cref="text"/>
-        /// </param>
+        /// <param name="text">Text content to split into lines.</param>
+        /// <param name="width">Maximum number of characters allowed per line.</param>
         /// <returns>
-        /// Collection of lines of at most <see cref="maxLength"/> characters
-        /// generated from the supplied <see cref="text"/>
+        /// Collection of lines of at most <paramref name="width"/> characters
+        /// generated from the supplied <paramref name="text"/>.
         /// </returns>
-        protected virtual IReadOnlyCollection<string> SplitText(string text, int maxLength)
+        protected virtual ReadOnlyCollection<string> SplitText(string text, int width)
```

Comment:
What's the reason for changing this to a concrete type?

If it's useful to do this, we could also make this method private.

Reply:
I made it concrete to follow the [guidelines for collections](https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/guidelines-for-collections).

I can return an interface but I need indexed access so I would have to return either `IList` or `IReadOnlyList`.

Reply:
`IReadOnlyList` is good.